code/game/turfs/open/_open.dm

```diff
@@ -187,10 +188,9 @@
 
 /turf/open/proc/freeze_turf()
 	for(var/obj/I in contents)
-		if(I.resistance_flags & FREEZE_PROOF)
-			continue
-		if(!(I.obj_flags & FROZEN))
-			I.make_frozen_visual()
+		if(!HAS_TRAIT(I, TRAIT_FROZEN) && !(I.obj_flags & FREEZE_PROOF))
```

Comment:
an I dumb? where is TRAIT_FROZEN used?

Reply:
freeze_turf() and /obj/item/freeze_cube/proc/freeze()

Reply:
wait i see what you mean, i never add the trait! not very good, i explicitly remember encountering a bug with the frozen element being added multiple times in the tm and i thought i fixed it there but it repeated after merge